### PR TITLE
Resolving GridUI caption error

### PR DESCRIPTION
**Pull request by ganimtron-10**
Fixing Bug mentioned at #463.

### CODE REVIEW

**Review by skoudoro — COMMENTED** [review]
Hi @ganimtron-10,

Thank you for doing that! Could you add a test also? Thank you

@antrikshmisri, Can you test this PR and check if it fix the reported issue. Thanks!

**Comment by ganimtron-10** [comment]
I have added tests for the changes.

**Comment by skoudoro** [comment]
Thank for the update @ganimtron-10!

ping @antrikshmisri

File: fury/tests/test_actors.py [hunk 1]
```diff
@@ -1176,6 +1176,7 @@ def timer_callback(_obj, _event):
     arr[arr < 20] = 0
     report = window.analyze_snapshot(arr)
     npt.assert_equal(report.objects, 6)
+    npt.assert_equal(type(container._items[0]), actor.Container)
```
Comment (antrikshmisri) [inline fury/tests/test_actors.py:1179]: Please avoid using private members, instead use the property.

File: fury/tests/test_actors.py [hunk 2]
```diff
@@ -1196,6 +1197,7 @@ def timer_callback(_obj, _event):
     arr = window.snapshot(scene)
     report = window.analyze_snapshot(arr)
     npt.assert_equal(report.objects > 6, True)
+    npt.assert_equal(type(container._items[0]), actor.Container)
```
Comment (antrikshmisri) [inline fury/tests/test_actors.py:1200]: Same comment as above.

**Review by antrikshmisri — COMMENTED** [review]
Hey @ganimtron-10, please see my comments below. Rest everything seems to be working fine. Thanks!

**Comment by ganimtron-10** [comment]
Thanks @antrikshmisri for the review!
I have made changes according to the comments.

File: fury/tests/test_actors.py (hunk 1 above)
Comment (ganimtron-10) [reply fury/tests/test_actors.py]: Done!

**Comment by antrikshmisri** [comment]
Hey @ganimtron-10, thanks for the update. LGTM.

**Review by antrikshmisri — APPROVED** [review]

**Comment by ganimtron-10** [comment]
Hii @antrikshmisri !! 
Is there any thing left to be changed in this pr??

**Review by skoudoro — COMMENTED** [review]
Thanks @antrikshmisri  for the review. 

I need to test this PR and we can go ahead and merge it. It should be done by the end of this week.

**Comment by ganimtron-10** [comment]
Hello @skoudoro ,
Any updates on this?

**Review by skoudoro — CHANGES_REQUESTED** [review]
Hi @ganimtron-10,

I do not think this is the right approach. 

It should be fixed at `GridUI` class level and not at the `grid`function level. 

Currently, you are adding complexity which is not necessary. to be more precise, you are doing:  `Container(Container(actor), Container(actor), etc...)`  and each container has only one actor so useless.

Please, let me know if you have time to update this PR, otherwise, I will close it. 

Thanks!

**Comment by ganimtron-10** [comment]
Hello @skoudoro ,
Thanks for the update!

Last time, I just got confused between the return type of the `grid` function and the type of the `items` as I thought the `grid` function is inconsistent (ie. which returns either `Container` or `Actor`) due to which I implemented that approach, but this isn't the case.

Now I have updated the PR and made changes into the `GridUI` itself.
PTAL.

**Review by skoudoro — COMMENTED** [review]
Much better! Thank you for the quick update! 

Can you just update the test of `gridUI`?  Then, it will be ready to merge. You could start with the initial issue #463 and make sure that we have the expected output in the test. 

Thanks !

**Comment by skoudoro** [comment]
@ganimtron-10, this PR is almost ready to be merged. 

See my comment above. Can you update the tests? Thank you!

**Comment by ganimtron-10** [comment]
Hello @skoudoro ,
I tried creating a similar test and as per the older conversation used try and except block to test whether it throw error or not. Please let me know if its correct or not.

**Comment by ganimtron-10** [comment]
Hey @skoudoro ,
I have updated the tests and they are now passing on `macOS`.

**Comment by skoudoro** [comment]
Thank you for this @ganimtron-10, looks good!

**Pull request merged**